### DISCUSSION TITLE
Windows 7 issues with slingox_server.exe

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#240] [General]
**NOTE: The simplest Win7 solution is just to install Python 3.8 and run the slingbox_server.py script; DO NOT try to run slingbox_server.exe, for all of the reasons outlined below**

This is a heads up to anyone who tries to run the server executable (slingox_server.exe) on a Windows 7 machine. (NOTE: the Python version of the server, slingox_server.py, works just fine.)

If you try to run the EXE on a Win7 machine as I did, you will end up with an error message that looks similar to one of the following, depending on how the EXE is launched:
* The program can't start because api-win-core-path-l1-1-0.dll is missing from your computer.
* Error loading Python DLL 'C:\Users\<username>\AppData\Local\Temp_MEI56002\python310.dll'. LoadLibrary: The specified module could not be found.

Discussion https://github.com/GerryDazoo/Slinger/discussions/213 provides some more details.

Both of these error messages stem from the fact that the Slinger EXE code was compiled using Python v3.10 libraries, and Python 3.10 is not supported on Windows 7. The highest version of Python supported on Windows 7 is 3.8.

For various reasons, the Python developers removed the api-win-core-path-l1-1-0.dll from Python distributions, starting in Python v3.9. For some insight into their reasons for doing this, and several proposals for how to "fix" this for Windows 7 users (which were all ultimately rejected by the Python developers), see "Windows 7 support for Python 3.9 or 3.10, without a fork" (https://discuss.python.org/t/windows-7-support-for-python-3-9-or-3-10-without-a-fork/13842).

Luckily, a resourceful individual took it upon themselves to implement one of the proposed solutions - much like Gerry has done with respect to Sling dropping support for Slingboxes. The resulting github project is titled "Unofficial Python Installers for Windows 7 and Windows Server 2003 R2" (https://github.com/adang1345/PythonWin7). Essentially, this project provides unofficial Win7 installers, both 32-bit and 64-bit, for all 3.9 and 3.10 versions of Python.

Frankly, the easiest deployment of Slinger on a Win7 machine is simply to use the Python version directly, slingox_server.py. But if you feel the need or desire to run the EXE version, you will almost certainly need to use one of the unofficial Python 3.10 installers from the github project mentioned in the prior paragraph.

**Comment by deputylynch** [comment]
So what 1 did you go with to use with Windows 7, 64 bit and how did you get it to run  in the slingbox software? 
I clueless and can't get this to run on My Old 7 machine with the same problems you had in you other post and don't know if I'm going to figure out how to get it to work!
I downloaded and installed 3.10.0 for 64bit and still can't figure how to implement it to work since I get the same error!

**Reply by DVoorhees1957** [reply]
I simply stopped trying to use the EXE file, I'm running the Python version (slingbox_server.py) directly. Does that not work for you? It should.

**Comment by deputylynch** [comment]
I have no idea how to do it. I've been trying to get it running for the last 30 hours!

**Comment by Lope0407** [comment]
I am attempting to build an exe that will work on Windows 7 - stand by...  Garry Lawson

**Reply by deputylynch** [reply]
For the time being if somebody could just explain to me how I could start it within the Python Application that will work would be great!

**Reply by DVoorhees1957** [reply]
If you've already downloaded and installed Python 3.8, which is the highest version of Python that can run on Windows 7, you just have to make sure that Python files (files with the ".py" extension) will be run by the Python executable. At a minimum, that means you have to add the Python directory that you installed Python 3.8 into (C:\Python38\ in my case) to the PATH environment variable. (I'm assuming you know how to modify the PATH environment variable - Google it if you don't know how.)

Then you just type the following command at a command prompt (assuming your config file is named config.ini):

slingbox_server.py config.ini

**Reply by deputylynch** [reply]
I'm sorry but I don't have a clue! I spent way too much time already trying to get the program to work and the reason I had a slingbox in the first place was because it was almost plug and play. Looks like I'm moving on since I picked up a used TiVo that looks like it's going to fulfill my needs even better than slingbox ever did! I got the system to run on a Windows 10 dedicated laptop and it's not all that solid in that it takes about five tries to get it to connect externally and I have to worry about the IP address  changing and the laptop being on all the time.
Let me know if you come up with an easier solution that will definitely work with the Windows 7 machine that I have on an old laptop that I was willing to dedicate full time. Thanks for your assistance!

**Reply by sp222222** [reply]
doesn't get any easier than what DV1957 mentions above. I have SS running on win7 box with python 3.8 and yes I too had to modify the path statement of windows7 (be very careful when you mod that long path statement. one wrong semicolon or removal of things can render the PC to not boot when rebooted).

**Comment by bradut** [comment]
One easy way to run Slinger on a **Windows 7** computer is to install there the **[Virtual Box](https://www.virtualbox.org/wiki/Downloads)** Virtual Machine, install Windows 10 in the VM, and set the VM's network adapter as "Attached to Bridge Adapter".

You will be able to run **slingbox_server.exe** in the Windows 10 instance. (So, no more Python compatibility issues, compilation, etc.) 
(Tried this a few minutes ago). 

Re: @deputylynch @sp222222 @DVoorhees1957 @Lope0407

**Reply by DVoorhees1957** [reply]
Thanks for a viable workaround! An even simpler solution would be to just install Python in Windows 7, and use the slingbox_server.py script rather than banging your head against the wall trying to use slingbox_server.exe. :-)
